File: java/src/main/java/ps_traning/programmers/level_3/최고의_집합.java
package ps_traning.programmers.level_3;

import java.util.Arrays;

public class 최고의_집합 {
    public int[] solution(int n, int s) {
        int[] answer = {-1};
        if ((int) (s / n) == 0) return answer;

        answer = new int[n];
        Arrays.fill(answer, (int) (s / n));
        int index = n - 1;
        for (int i = 0; i < s % n; i++) {
            answer[index--]++;
        }
        return answer;
    }
}

File: python/with_java/programmers/level 3/최고의 집합.py
def solution(n, s):
    if s // n == 0:
        return [-1]
    answer = [s // n for _ in range(n)]
    per = s % n
    i = n-1
    for _ in range(per):
        answer[i] += 1
        i -= 1
    return answer